fms2_io/fms_netcdf_domain_io.F90

```diff
@@ -546,44 +546,59 @@ subroutine save_domain_restart(fileobj, unlim_dim_level)
   if (.not. fileobj%is_restart) then
     call error("file "//trim(fileobj%path)//" is not a restart file.")
   endif
+
+! Calculate the checksum and write it out
   do i = 1, fileobj%num_restart_vars
-    if (associated(fileobj%restart_vars(i)%data0d)) then
-      call domain_write_0d(fileobj, fileobj%restart_vars(i)%varname, &
-                           fileobj%restart_vars(i)%data0d, unlim_dim_level=unlim_dim_level)
-    elseif (associated(fileobj%restart_vars(i)%data1d)) then
-      call domain_write_1d(fileobj, fileobj%restart_vars(i)%varname, &
-                           fileobj%restart_vars(i)%data1d, unlim_dim_level=unlim_dim_level)
+    if (associated(fileobj%restart_vars(i)%data0d) .and. associated(fileobj%restart_vars(i)%data1d)) then
```

Comment:
I don't think this is right.  You should never have both data0d and data1d associated at the same time.  I think a better solution is to delete this entire branch of the if statement, and then delete the else branch on line 575.